aws/amazon-chime-sdk-android
# Provide default device api

### Issue #, if available:
Chime-31100
### Description of changes:
This is to add getActiveAudioDevice API. Due to behavior change, we decide to use Android API to get device from Android API which is supported above >= 24.

Android provides some of API [`isWiredHeadSetOn`](https://developer.android.com/reference/android/media/AudioManager#isWiredHeadsetOn()), but this doesn't reflect actual device that is playing the sound, so we are not able to rely on to these APIs.

### Testing done:
1. Connect bluetooth earphone. If connected, it should show bluetooth earphone as active device
2. Connect bluetooth earphone. If failed, it should show current active device

#### Unit test coverage
* Class coverage: 85%
* Line coverage: 77%

#### Manual test cases (add more as needed):
* [X] Join meeting
* [X] Leave meeting
* [X] See active speaker indicator when speaking
* [X] Mute/Unmute self
* [ ] See roster updates when remote attendees join / leave the meeting
* [ ] Enable local video
* [ ] See local video tile
* [ ] See remote video tile
* [ ] Pause remote video tile
* [ ] Resume remote video tile
* [ ] First time audio permissions
* [ ] First time video permissions

#### Screenshots, if available:
![image-1fecd05a-b05c-49e4-976f-00447ed6b425](https://user-images.githubusercontent.com/57926812/94897348-482b0c00-0444-11eb-9431-e3c4c52123ca.jpg)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
